test/ttt_test.exs

```diff
@@ -3,38 +3,56 @@ defmodule TTTTest do
   import ExUnit.CaptureIO
   doctest TTT
 
-  test "Game will automatically switch turn to 'o' after 'x' has played and vice versa." do
+  test "Game will automatically switch turn to 'o' after 'x' has played and vice versa" do
     first_player = :x
     assert TTT.switch_turn(first_player) == :o
 
     first_player = :o
     assert TTT.switch_turn(first_player) == :x
   end
 
-  test "Only a string of '1' to '9' will be accepted as valid input." do
-    input1 = "1"
-    input2 = " "
-    input3 = "aa"
-    input4 = ",,"
-    input5 = "10"
-    input6 = "0"
-    input7 = "5"
-    assert TTT.valid_input?(input1) == :true
-    assert TTT.valid_input?(input2) == :false
-    assert TTT.valid_input?(input3) == :false
-    assert TTT.valid_input?(input4) == :false
-    assert TTT.valid_input?(input5) == :false
-    assert TTT.valid_input?(input6) == :false
-    assert TTT.valid_input?(input7) == :true
+  test "A number '1' through '9' will be a valid input" do
+    inputs = ["1", "2", "5", "8", "9"]
+    for input <- inputs, do: assert TTT.valid_input?(input) == true
+  end
+
+  test "Whitespaces or newlines before or/and after a valid input will be ignored, such input will be a valid one" do
+    inputs = ["  3  ", "    7", "4 ", "\n2\n\n"]
+    for input <- inputs, do: assert TTT.valid_input?(input) == true
+  end
+
+  test "A number smaller than '1' or greater than '9' will be invalidated" do
+    invalids = ["-1", "-23", "0", "10", "50", "120"]
+    for input <- invalids, do: assert TTT.valid_input?(input) == false
+  end
+
+  test "Symbols will be invalidated" do
+    symbols = ["`", "~", "@", "#", "$"]
+    for symbol <- symbols, do: assert TTT.valid_input?(symbol) == false
+  end
+
+  test "Letters will be invalidated" do
+    letters = ["a", "z", "G", "T"]
+    for letter <- letters, do: assert TTT.valid_input?(letter) == false
+  end
+
+  test "Empty input, whitespaces and newlines will be invalidated" do
+    inputs = ["", " ", "   ", "\n", "\n\n\n"]
+    for input <- inputs, do: assert TTT.valid_input?(input) == false
+  end
+
+  test "Other characters will be invalidated" do
+    inputs = ["哈", "ト", "قلب", "হৃদয়", "หัวใจ"]
+    for input <- inputs, do: assert TTT.valid_input?(input) == false
   end
 
   test "Valid user input will be map to the correct coordinate on the board" do
-    TTT.match_input("1") == {0, 0}
-    TTT.match_input("9") == {2, 2}
-    TTT.match_input("5") == {1, 1}
+    assert TTT.match_input("1") == {0, 0}
+    assert TTT.match_input("9") == {2, 2}
+    assert TTT.match_input("5") == {1, 1}
   end
 
-  test "Visual board is updated accordingly" do
+  test "Board is updated and rendered correctly" do
```

Comment:
"Board is updated and rendered correctly" - This test name does not give me much in the way of details. Under what conditions is this true? Does it ever throw an exception or fail? A better name might be:

"Board is updated and rendered correctly when user provides valid input"

Or something like that